lib/resty/core/regex.lua

```diff
@@ -153,6 +154,14 @@ local function get_max_regex_cache_size()
     return max_regex_cache_size
 end
 
+function _M.regex_cache_is_empty()
+    if not regex_match_cache then
+        return true
+    else
+        return next(regex_match_cache.hasht) == nil
```

Comment:
@alubbe The simplest thing for us to do is to simply set a flag when any of those `ngx.re.*` API functions that may cache a new compiled regex are called.